lua/core/menu/tape.lua

```diff
@@ -70,16 +72,25 @@ local function update_tape_index()
   if f == nil then
     print("WARNING: couldn't write index file, even after creating `tape/`")
   else
-    f:write(tostring(m.fileindex+1))
+    f:write(tostring(v))
     f:close()
   end
 end
 
+local function update_tape_index()
+  read_tape_index()
+  write_tape_index_v(m.fileindex+1)
+end
+
 local function edit_filename(txt)
   if txt then
     m.rec.file = txt .. ".wav"
   else
-    m.rec.file = string.format("%04d",m.fileindex) .. ".wav"
+    _menu.redraw()
+    return
+  end
+  if txt == string.format("%04d",m.fileindex+1) then
+    update_tape_index(v)
```

Comment:
NB: we use `update_tape_index()` instead of `write_tape_index_v(txt)`.
so it re-reads index value on disk before updating, which can be nice to handle a tad better **concurrent writes**.

still, **can concurrent writes actually happen (triggered by scripts)?**

even if this way of doing provides very basic protection against some LLW cases (last write wins), it's not that robust.
i could implement a strong serializable guarantee w/ a loop (given that `fileindex` can basically work as a [Lamport timestamp](https://en.wikipedia.org/wiki/Lamport_timestamp) but do we want to go there?

Reply:
it'd be good to have sane protections. yes, all tape functions can be controlled via scripts via https://github.com/monome/norns/blob/main/lua/core/audio.lua#L147

Reply:
i was gonna PR something like

```lua
local function update_tape_index()
  local want_to_write = m.fileindex + 1
  read_tape_index() -- updates m.fileindex for next step
  while m.fileindex < want_to_write do
    write_tape_index_v(want_to_write)
    read_tape_index() -- read our own write case concurrent write happened
  end
end
```

but then realized that (even w/ coroutines) norns Lua env is single-threaded, so the above protection should be sufficient.